tests/python_tests/samples/test_speculative_decoding_lm.py

```diff
@@ -45,3 +45,42 @@ def test_sample_speculative_decoding_lm(self, convert_model, convert_draft_model
         assert cpp_result_ref.stdout.strip() in py_result.stdout.strip(), "Python and CPP results should match"
         assert cpp_result_ref.stdout.strip() in cpp_result.stdout.strip(), "Greedy and speculative decoding results should match"
 
+test_prompt = """Code:
+def add(a, b):
+    return a + b
+
+Question: Can you please add 2 and 3
+A:"""
+class TestEagle3SpeculativeDecodingLM:
+    @pytest.mark.llm
+    @pytest.mark.samples
+    @pytest.mark.parametrize(
+        "convert_model, convert_draft_model, sample_args",
+        [
+            pytest.param("Qwen3-1.7B", "qwen3_1.7b_eagle3", test_prompt, marks=pytest.mark.skip(reason = 'CVS-171947, CVS-171943, CVS-174959')),
+        ],
+        indirect=["convert_model", "convert_draft_model"],
+    )
+    def test_sample_speculative_decoding_lm(self, convert_model, convert_draft_model, sample_args):
```

Comment:
please, parameterize the TestSpeculativeDecodingLM test or move the common code into a separate function